src/code/z_scene.c

```diff
@@ -3,116 +3,133 @@
+ * When an object is spawned with this function, all objects that come before it in the entry list will be treated as
+ * persistent, which will likely cause either the amount of free slots or object space memory to run out.
+ * This function is only meant to be called internally on scene load, before the object list from any room is processed.
+ */
+s32 Object_SpawnPersistent(ObjectContext* objectCtx, s16 objectId) {
     u32 size;
 
-    objectCtx->status[objectCtx->num].id = objectId;
+    objectCtx->slots[objectCtx->numEntries].id = objectId;
     size = gObjectTable[objectId].vromEnd - gObjectTable[objectId].vromStart;
 
-    osSyncPrintf("OBJECT[%d] SIZE %fK SEG=%x\n", objectId, size / 1024.0f, objectCtx->status[objectCtx->num].segment);
+    osSyncPrintf("OBJECT[%d] SIZE %fK SEG=%x\n", objectId, size / 1024.0f,
+                 objectCtx->slots[objectCtx->numEntries].segment);
 
-    osSyncPrintf("num=%d adrs=%x end=%x\n", objectCtx->num, (s32)objectCtx->status[objectCtx->num].segment + size,
-                 objectCtx->spaceEnd);
+    osSyncPrintf("num=%d adrs=%x end=%x\n", objectCtx->numEntries,
+                 (s32)objectCtx->slots[objectCtx->numEntries].segment + size, objectCtx->spaceEnd);
 
-    ASSERT(((objectCtx->num < OBJECT_EXCHANGE_BANK_MAX) &&
-            (((s32)objectCtx->status[objectCtx->num].segment + size) < (u32)objectCtx->spaceEnd)),
+    ASSERT(((objectCtx->numEntries < ARRAY_COUNT(objectCtx->slots)) &&
+            (((s32)objectCtx->slots[objectCtx->numEntries].segment + size) < (u32)objectCtx->spaceEnd)),
            "this->num < OBJECT_EXCHANGE_BANK_MAX && (this->status[this->num].Segment + size) < this->endSegment",
            "../z_scene.c", 142);
 
-    DmaMgr_SendRequest1(objectCtx->status[objectCtx->num].segment, gObjectTable[objectId].vromStart, size,
+    DmaMgr_SendRequest1(objectCtx->slots[objectCtx->numEntries].segment, gObjectTable[objectId].vromStart, size,
                         "../z_scene.c", 145);
 
-    if (objectCtx->num < OBJECT_EXCHANGE_BANK_MAX - 1) {
-        objectCtx->status[objectCtx->num + 1].segment =
-            (void*)ALIGN16((s32)objectCtx->status[objectCtx->num].segment + size);
+    if (objectCtx->numEntries < (ARRAY_COUNT(objectCtx->slots) - 1)) {
+        objectCtx->slots[objectCtx->numEntries + 1].segment =
+            (void*)ALIGN16((s32)objectCtx->slots[objectCtx->numEntries].segment + size);
     }
 
-    objectCtx->num++;
-    objectCtx->unk_09 = objectCtx->num;
+    objectCtx->numEntries++;
+    objectCtx->numPersistentEntries = objectCtx->numEntries;
 
-    return objectCtx->num - 1;
+    return objectCtx->numEntries - 1;
 }
 
-void Object_InitBank(PlayState* play, ObjectContext* objectCtx) {
-    PlayState* play2 = play; // Needs to be a new variable to match (possibly a sub struct?)
+// 1000 kilobytes
+#define OBJECT_SPACE_SIZE_BASE (1000 * 1024)
+
+void Object_InitContext(PlayState* play, ObjectContext* objectCtx) {
```

Comment:
`Object_InitContext` doesn't seem great since it makes it sound like the function only initializes the context struct, and it doesn't make it clear that it also allocates the space and spawns the initial object(s), which is just as important if not more imo.
However I don't really have a suggestion either, aside from referring to the space and system as a whole in some way (ie. the "bank" in the previous naming)

Reply:
-> `Object_Init` I guess

Reply:
`Object_Init` sounds very confusing since it intializes the whole space, and not a specific object, which kinda goes against how we use `<Thing>_Init` in the rest of the codebase (eg. <Actor>_Init)

Reply:
`Interface_Init`, `Message_Init`, `Map_Init`, `MapMark_Init`, `Skybox_Setup`, `Skybox_Init` use `GameState_Alloc` and load stuff then, maybe `Object_Setup` ? `Object_SetupContext` ?

Or there is `func_80097dd8`, there are more Object_ system functions to name anyway, and the other function `func_80096FE8` needs a name but also has `GameState_Alloc`/load data combo and I don't think it'll work as a `_Init` (the call to it is also weirdly hidden in a printf...)

Reply:
we can also change the system name to "objects" since it handles several objects anyway. `Objects_Init`

Reply:
`Object_Setup` has the same issue as `Object_Init` imo, it sounds like it does the work needed for an actual "object". And `Object_SetupContext` isn't really better than `Object_InitContext` in that the function does more than just context initialization or setup. Tho I do think `Object_InitContext` is better than `Object_Init`/`Object_Setup`, fwiw

Among the init functions you mentioned, almost all are init functions for a system where there is no ambiguity between initializing the system or one of the instances it manages (ie. there is only 1 "interface" or 1 "skybox" at a time, and that's what the systems manage), so the names work well. The only exception would be `Message_Init`, since you might mistake it as initializing a specific text aka. a message, and I would argue it's also misleading to some extent (tho we primarily use "text" in our names so it might be ok).
But for a system like objects, I think we should use the same distinction made in the actor or effect systems for instance, where we use names like `Actor_UpdateAll` to differentiate from `Actor_Update`

Reply:
there is `Object_UpdateEntries`, `Object_InitEntries` ?

Reply:
It doesn't initialize the entries either tho, it just intializes the space (aka. the bank).

And to be clear I don't think `Object_InitContext` is unusable either, it's probably the best among everything that's been suggested so far. I just think it's a downgrade over a clear name like `Object_InitBank` (tho you can change "bank" by whichever equivalent term you want to replace it with). And I feel like we're going to have similar issues when naming the rest of the system in the future, but that might just be me

Reply:
the main logic is in `Scene_CommandObjectList` (popping the slots from the first mismatch between the two object lists), the few functions left are unrelated to the memory management aspect afaict (well, one is 100% memory management but just loading back objects that were overwritten by kaleidoscope, to exactly where they were before)

A critic of the suggested change could also be that `ObjectEntry slots[19];` looks weird, maybe `ObjectEntry`'s name should change but I can't find a much better name, and the stuff using "entry" is internal anyway.
The main point with changing objBankIndex to objectSlot is to get a unique and less confusing (objBankIndex // objIndex) way to refer to these entries and externally only `ObjectContext.slots` is used
(that confusion was also a straight path from the devs' names apparently:
https://github.com/zeldaret/oot/blob/2e18eb334bad77db0625549c62849333705214b8/src/code/z_actor.c#L2747-L2749
this debug strings calls an objectId "bank", this do be great naming)

Reply:
I think regardless of how the original game is using the term bank, we can just use it as a term to describe the thing that holds all of these entries. Or we can use the term "pool", we just need something. Point is its the name of the thing that has slots that you have these load entries in

Reply:
*Deletes comment to avoid getting in the "bank" argument*

Did we rule out `Object_InitSpace` ?

Reply:
`Object_Allocate` ? c.f. `BgCheck_Allocate`

Reply:
`Object_Allocate` on the surface sounds like its allocating a single object. even tho of course thats not how the system works at all, I think it gives the wrong impression. and the function is doing more than just allocating some space anyway

`Object_InitSpace` is a step in the right direction I suppose, but I think i do agree with roman here that it needs some sort of concrete name. "space" isnt giving a name to the thing thats holding the slots, its just referencing to this chunk of memory that this thing will go. So it doesnt feel 100% right to me

I honestly dont see what the huge issue is with bank. The bank can exist as its own entity that has slots like we are currently calling them

Reply:
We currently have the object context, which manages the object space by dividing it between slots

I think that's enough words and terminology to describe the system, I don't see what a single use of "bank" would refer to
That's why I thought InitContext or InitSpace would work, they're established and well-used terms within the system, countrarily to Bank

To summarize, if I read "init bank" in the object system I'd be like "what is bank"

Reply:
Been reading/thinking through this for a while now.
Currently it seems like `space` is already kind of used for this terminology referring to the full object space (based on the struct members in ObjectContext `spaceStart` and `spaceEnd`, and the stack variable `spaceSize` (unless these are for something else)). Based on that I think `InitSpace` would be fine. Though I think `bank` or `pool` also makes sense for this terminology referring to the full object space. Perhaps we could just use `slots` to refer to this space as well?

Reply:
> Perhaps we could just use slots to refer to this space as well?

I think its important to not conflate the actual space/buffer where the objects go and the slots. The slots are definitely separate, and the amount of slots and their existence doesnt really have any bearing on the buffer and how big it is. the 19 slots oot has is pretty arbitrary, and isnt really whats getting initialized here

Reply:
as for InitSpace, it def makes sense based on how we are calling this space with variables like `spaceStart` and `spaceEnd`, but something just sounds weird about init space lol
Something like bank feels a little more concrete, but has been a contentious term to use so far. 
Maybe just buffer since thats all it really is. InitBuffer/AllocateBuffer, idk

Reply:
`InitBuffer` is probably more appropriate since on top of allocating the space it also spawns gameplay keep

Reply:
I don't have much to add to https://github.com/zeldaret/oot/pull/1227#discussion_r963341163 (I have the same objections to Buffer as Bank, in the sense that if I was to agree to Buffer I would also agree to Bank)

How I see it is `InitSpace` would initialize the space (allocate the memory) and its partition (resets the slots), and start filling it by spawning gkeep

Reply:
oh, yeah I did forget that there is a loop that set the slot ids to OBJECT_INVALID.

I dont think buffer has the same issues/implications as bank though. buffer tells it like it is, its a raw buffer with no sort of managment system (ie its not a heap/linked list/stack or anything). Whereas bank is a term that doesnt really give you any information, and as you say in that old comment, you ask youself "what is the bank". buffer is pretty clear, its the buffer where the files themselves go.

Reply:
Personally I think `Object_InitContext` is a fine name, it isn't like we can add all the stuff a function does in the name itself. Something similar happens on `Actor_InitContext` where that function also resets the loaded entries of the overlay table, initialize the target context and the farore's wind stuff
I think it could be noted as a comment what this function does, like allocating space and initializing objects.